service/history/workflow/mutable_state_impl.go

```diff
@@ -288,10 +288,11 @@ func NewMutableState(
 
 		LastCompletedWorkflowTaskStartedEventId: common.EmptyEventID,
 
-		StartTime:              timestamppb.New(startTime),
-		VersionHistories:       versionhistory.NewVersionHistories(&historyspb.VersionHistory{}),
-		ExecutionStats:         &persistencespb.ExecutionStats{HistorySize: 0},
-		SubStateMachinesByType: make(map[int32]*persistencespb.StateMachineMap),
+		StartTime:                         timestamppb.New(startTime),
+		VersionHistories:                  versionhistory.NewVersionHistories(&historyspb.VersionHistory{}),
+		ExecutionStats:                    &persistencespb.ExecutionStats{HistorySize: 0},
+		SubStateMachinesByType:            make(map[int32]*persistencespb.StateMachineMap),
+		TaskGenerationShardClockTimestamp: shard.CurrentVectorClock().GetClock(),
```

Comment:
NewMutableStateFromDB invokes NewMutableState... 
need to set this field back to its persisted value in NewMutableStateFromDB

Reply:
It's part of ExecutionInfo, so it should be populated automatically.

Reply:
oh u r right. we overwrite the entire executionInfo.